manager/src/agent.rs

```diff
@@ -119,14 +146,46 @@ impl Contract {
             );
             let withdrawal_amount = agent_balance - storage_fee;
             agent.balance = U128::from(agent_balance - withdrawal_amount);
-            self.agents.insert(&account, &agent);
+
+            // if this is a full exit, remove agent. Otherwise, update agent
+            if remove.is_some() && remove.unwrap() {
```

Comment:
I've never seen this used for a conditional, but looks fine. I almost always see the `if let Some(b)…`

Reply:
good call